### ISSUE TITLE
Experimental label on Idea Hub module

### ISSUE THREAD

**Issue by eclarke1** [google/site-kit-wp#3810]
## Feature Description

Addition of a label/tag on the Idea Hub module/widget on the Site Kit dashboard to indicate it is 'Experimental' once Idea Hub is activated.

Additional consideration is required for other places this label/tag or additional copy is required throughout the SK dashboard and WordPress, for active and inactive states of Idea Hub, such as:

- CTA banner/notification in SK dashboard header (when Idea Hub is inactive)
- Widget CTA on SK dashboard (next to Top Keywords widget) (when Idea Hub is inactive)
- SK settings page (both when Idea Hub is active and inactive)
- WP dashboard SK widget (when Idea Hub is active)
- WP posts screen (when Idea Hub is active)

We may not require all of the above, but ACs to consider each.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

1. **Widget CTA on the main dashboard**
    - update heading to: _"Get new topics to write about"_
    - add text (same font size as the current "learn more", but in regular text color) after the lightbulb : _"Idea Hub is an experimental new feature that shows you suggestions to write about based on the content of your site"_
    - keep the "learn more" link in blue after the text
2. **Setup screen** (with tracking opt-in checkbox)
    - add "experimental" in small caps after the Idea Hub name 
    - update the opt-in checkbox sentence so all verbs are in the same form: _"creating a draft; saving/dismiss**ing** ideas"_
3. **Connected Services screen in Settings** (after setup is complete and the service is connected) 
    - add "experimental" in small caps after the Idea Hub name 
4. **Connect More Services screen in Settings** (if the feature is not activated)  
    - add "experimental" in small caps after the Idea Hub name 
5. **Main dashboard widget** (once the feature is activated)
    - add "experimental" in small caps after "Ideas to write about based on unanswered searches"

A new `Badge` React component should be introduced to render the "experimental" label from the points 2-5 above with consistent markup and styling.

## Implementation Brief
* Create `assets/js/components/Badge.js` which exports the `Badge` functional component.
	* It should accept the `label` prop which is a required string.
	* It should render the `label` between `span` tags and having the `googlesitekit__badge` class name.
* Create `assets/sass/components/global/_googlesitekit-badge.scss` which contains the styling for the `Badge` component. Refer to the following [Figma designs](https://www.figma.com/file/YB79eKcGOwyUjlWG8mjDZA/Prompt-Engine-%2F-Idea-Hub?node-id=21114%3A10483) for styling directions.
* Using `assets/js/modules/idea-hub/components/dashboard/DashboardCTA.js`,
	* Update the text in the heading from `Get new topics based on what people are searching for with Idea Hub` to the one defined in the AC.
	* Remove the small light bulb and the corresponding SVG file
	* Before the "Learn More" text, include the text as per the AC having the same font size as "Learn More" but in regular text color.
* Using `assets/js/modules/idea-hub/components/setup/SetupMain.js`,
	* Render `Badge` after the module title but within the module title tag.
* Using `assets/js/modules/idea-hub/components/setup/SetupForm.js`,
	* Update `creating a draft; saving/dismiss ideas` to the text defined in the AC.
* Using `assets/js/components/settings/SetupModule.js` and `assets/js/components/settings/SettingsActiveModule/Header.js`,
	* Display `Badge` after the module name if the module being displayed is `Idea Hub` by checking for the `slug`.
* Using `assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js`,
	* Include `Badge` after "Ideas to write about based on unanswered searches"
* In all the above bullet points, the `label` prop should be passed to the `Badge` component, with the translated `Experimental` string as value.

### **Draft PR:**
https://github.com/google/site-kit-wp/pull/3900

### Test Coverage
* No tests to be added

### Visual Regression Changes
N/A

## QA Brief

* Follow the acceptance criteria above, and verify that the text and / or badge is correct on the various Idea Hub screens mentioned.
* In addition, note the discussion below regarding the removal of the bulb icon from the dashboard CTA.

## Changelog entry

* Add "Experimental" badge to Idea Hub UI.

**Comment by felixarntz** [comment]
@aaemnnosttv For the bullet points 2-5 of the ACs: Do you think we should hard-code this considering this a special case? In principle it would be nice to have it tie into the module registration (e.g. `experimental: true`), but I also think this is very specific and we don't know what else could be relevant like this, e.g. would we rather want a boolean `experimental` or a string `annotationText`?

**Comment by aaemnnosttv** [comment]
> Do you think we should hard-code this considering this a special case?

@felixarntz yes, I think it doesn't make sense to build it in to any APIs just yet. I think we can add a component for this to keep things consistent but otherwise, I think it makes sense to handle it inline for now. If anything we could add a new generic "badge" component for this kind of visual element that could be reused in other ways and places to avoid it being entirely unique.

**Comment by felixarntz** [comment]
@aaemnnosttv Sounds good, I've added the requirement of a `Badge` component to the ACs.

**Comment by asvinb** [comment]
@aaemnnosttv For #2 in the AC, do you know exactly where we should put the experimental label? Because there are many "Idea Hub" text on that screen.
![image](https://user-images.githubusercontent.com/31818/129194649-f1982252-8837-4a2f-9df5-0ac3a82dfa63.png)

Also am wondering whether we can pass an experimental flag to the module when actually registering the module. Thus if we don't want to display the experimental badge, for e.g the settings screens where we check for the module slug, we just need to update the flag when the module is registered. True there are other places where the text is hard coded but maybe it's good after all to know somewhere in the code that this module right now is experimental. What do you think?

**Comment by felixarntz** [comment]
@asvinb The "experimental" label should appear behind the "Idea Hub" heading on the setup screen.

Regarding registering the module as experimental, Evan and I discussed this earlier on the issue, but decided against it. This isn't something we want to advocate for or encourage - it should be implemented specifically for the Idea Hub module for now, and it is acceptable to have specific "if Idea Hub" clauses in a few places where unavoidable.

**Comment by aaemnnosttv** [comment]
Thanks @asvinb! Per a group call earlier, it was deemed important that we have a design to work with here to minimize iterations later as the end result isn't clearly defined as-is.

@felix – for the badge I envision the visual treatment to be similar to a label in GitHub or Gmail.

Here's a quick mockup I made to show what I had in mind
![image](https://user-images.githubusercontent.com/1621608/129243410-13b61546-6e79-4c78-a6e2-0cff07cc23db.png)

@marrrmarrr mentioned that we probably don't want the "experimental" badge to be colorful, but I think we could have colorful badge variants for other things – e.g. "recommended" which I think we have in a few places like AdSense 
![image](https://user-images.githubusercontent.com/1621608/129243824-a3578ebb-6a0f-4175-b3fb-ff297591bc51.png)

The [lab beaker icon](https://fonts.google.com/icons?selected=Material+Icons:phone_android&icon.query=experiment) is also commonly used to show experimental status visually if we wanted to be more concise about it. E.g. 
![image](https://user-images.githubusercontent.com/1621608/129244511-2e76bfcb-2b5f-4078-ad7c-c68649e3054a.png)

**Comment by felixarntz** [comment]
@asvinb The IB looks good, just one thing to be refined:

> * Create `assets/js/components/Badge.js` which exports the `Badge` functional component.
>	* It should render the `experimental` string between `span` tags and having the `googlesitekit__badge--experimental` class name.

The `Badge` component itself should not have any association with the "experimental" label in itself. It should instead receive a string as prop, and we would render it passing the "experimental" string to the component. As such, the class to use for the component should also not contain the term "experimental", i.e. `googlesitekit-badge` would be sufficient.

@aaemnnosttv Regarding the styling, let's finalize that as part of the PR and review with @marrrmarrr and @Pratheep-lab. I think the first mockup you have above looks good for a start. What we need to ensure in the CSS is that it will align nicely regardless of the style of the text it is associated with (e.g. after a big heading vs regular text vs the module settings panel header).

**Comment by alex-moulin** [comment]
@felixarntz @aaemnnosttv @marrrmarrr following the request on yesterday's call, I put together a mockup to illustrate where the badge should be positioned according to the Acceptance Criteria above.

You can review the mockup here - https://www.figma.com/file/YB79eKcGOwyUjlWG8mjDZA/?node-id=21114%3A4117

Any questions/comments, please let me know.

**Comment by asvinb** [comment]
@felixarntz IB has been updated, however I left a couple of comments in Figma for @alex-moulin to review. Based on @alex-moulin 's feedback, we may need to update the estimate.

**Comment by alex-moulin** [comment]
@asvinb Happy with the suggestion to make things simpler to implement so I updated the wires accordingly.

**Comment by asvinb** [comment]
Thanks @alex-moulin !

**Comment by felixarntz** [comment]
@asvinb Two final things about the IB:

> It should accept the `label` prop, having an empty string as default value.

It would probably be more appropriate to mark this as required instead of using an empty default. A `Badge` without a label should not be possible to have.

> In all the above bullet points, the `label` prop should be passed to the `Badge` component, with `experimental` as value.

Maybe that's implied, but I want to make sure that "experimental" is specified as translatable string (as for all user-facing strings) and not just the string itself. Just double checking.

**Comment by asvinb** [comment]
Sounds good @felixarntz . I've updated the IB by adding the `label` prop as required. As for the `experimental` string, it's implied that it should be translatable, but i made it clear in the IB.
A couple of other things:

- after talking to @aaemnnosttv and @alex-moulin , we found out that the [small bulb icon](https://github.com/google/site-kit-wp/blob/develop/assets/js/modules/idea-hub/components/dashboard/DashboardCTA.js#L98) in the dashboard CTA widget is not adding much and looks out of place in different scenarios and we think it's better that we drop the bulb altogether. Here are the different scenarios:
	* Bulb centered: ![image](https://user-images.githubusercontent.com/31818/129878426-1dccd5d3-3252-4a13-8f2f-e087f4469e6c.png)
	* Bulb aligned top: ![image](https://user-images.githubusercontent.com/31818/129878489-b64e9cd9-b525-4e11-bd9c-6a42198107d0.png)
	* Inline bulb: ![image](https://user-images.githubusercontent.com/31818/129878556-b8466be4-3c43-461e-9f72-594d558726c3.png)
	* No bulb: ![image](https://user-images.githubusercontent.com/31818/129878629-2f51dbfc-37f3-49fe-870a-fb54658cf14b.png)
- @alex-moulin Also suggested we move the "Learn More" link to be next to the "Set Up" button as it would improve usability as otherwise the CTA are quite close to one another.

Let me know what you think, and I'll update the IB if you're fine with the above bullet points. There is also a [draft PR](https://github.com/google/site-kit-wp/pull/3900) where you can visually see the changes as well.

**Comment by felixarntz** [comment]
@asvinb Thanks, that looks good. Regarding your two additional points:
* Let's drop the bulb icon like you've discussed.
* For the "Learn more" link, I think that should remain right after the text as it is now, since this is a common Site Kit practice. If that is a concern, let's rather think about it holistically. cc @alex-moulin

**Comment by asvinb** [comment]
Sounds good @felixarntz . IB updated!

**Comment by alex-moulin** [comment]
@felixarntz Sounds like a plan

**Comment by aaemnnosttv** [comment]
@felixarntz @asvinb a few thoughts about the component:

- any reason why we wouldn't use `children` for the text/label? We could still make it required
- I realize we only have one badge so far, but regarding the component it might be a good idea to approach it with other states in mind for the future. Not that we need to implement any now, but that we may want to have variants in the future ([example](https://getbootstrap.com/docs/5.1/components/badge/#pill-badges)) The current light gray could be the default.

**Comment by felixarntz** [comment]
@aaemnnosttv I don't feel strongly about it, but I'd prefer to reserve `children` for things where we want to leave more flexibility. We probably don't want to allow for a badge to include e.g. bold or emphasized text within - the badge itself should fully control the visual appearance IMO. I'd prefer to leave `children` for things like potential multiline text or sentences where including formatting within would be more important.

@asvinb Let's follow up on the above, we can still make the final decision during execution. For now, IB ✅

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@asvinb An observation.  For the Widget CTA on the main dashboard:  The `learn more` link is external so should we have the icon displayed next to it for consistency? I realise it isn't in the AC but thought I'd raise it. 

**Verified:**

_Widget CTA on the main dashboard_ 

- The heading is updated to: `Get new topics to write about` along with the added text `Idea Hub is an experimental new feature that shows you suggestions to write about based on the content of your site` Also confirmed that the bulb no longer appears. 

![image](https://user-images.githubusercontent.com/73545194/131351287-c699ff58-7769-4005-abbd-928fefac06ae.png)

_Setup screen_

- The `experimental` label appears in small caps after the Idea Hub name.
- The opt-in checkbox sentence so all verbs are in the same form: creating a draft; saving/dismissing `ideas`

![image](https://user-images.githubusercontent.com/73545194/131352908-d2b71b06-a8ed-4d2a-85ac-d0f6edfd145e.png)

_Connected Services screen in Settings_

- The `experimental` label appears in small caps after the Idea Hub name

![image](https://user-images.githubusercontent.com/73545194/131353256-dd898b11-1621-48d6-bf1b-b7c01f72f5ff.png)

_Connect More Services screen in Settings_

- The `experimental` label appears in small caps after the Idea Hub name

![image](https://user-images.githubusercontent.com/73545194/131353816-71acf1d0-7b57-4b33-94ab-4e57fbad1fa9.png)

_Main dashboard widget_ 

The `experimental` label appearss in small caps after `Ideas to write about based on unanswered searches`

![image](https://user-images.githubusercontent.com/73545194/131354073-e32496f1-3a6f-42d3-83c3-277e22844436.png)

- Confirmed that the styling and markup is consistent between all of the areas where the new component appears. - [Screenshot](https://user-images.githubusercontent.com/73545194/131354893-2f41e18d-854f-474e-9909-3fb1ea64abcd.png)
- Checked the new component on supported browsers and on smaller screen sizes.

**Comment by tofumatt** [comment]
@wpdarren It looks like the `hideExternalIndicator` flag has been on that component for some time, but I don't see why it should be using that style.

I'll file an issue to remove it, but I think it should still pass QA here. Thanks! 👍🏻

**Comment by wpdarren** [comment]
QA Update:  ✅ 

**Verified:**

_Widget CTA on the main dashboard_ 

- The heading is updated to: `Get new topics to write about` along with the added text `Idea Hub is an experimental new feature that shows you suggestions to write about based on the content of your site` Also confirmed that the bulb no longer appears. 

![image](https://user-images.githubusercontent.com/73545194/131351287-c699ff58-7769-4005-abbd-928fefac06ae.png)

_Setup screen_

- The `experimental` label appears in small caps after the Idea Hub name.
- The opt-in checkbox sentence so all verbs are in the same form: creating a draft; saving/dismissing `ideas`

![image](https://user-images.githubusercontent.com/73545194/131352908-d2b71b06-a8ed-4d2a-85ac-d0f6edfd145e.png)

_Connected Services screen in Settings_

- The `experimental` label appears in small caps after the Idea Hub name

![image](https://user-images.githubusercontent.com/73545194/131353256-dd898b11-1621-48d6-bf1b-b7c01f72f5ff.png)

_Connect More Services screen in Settings_

- The `experimental` label appears in small caps after the Idea Hub name

![image](https://user-images.githubusercontent.com/73545194/131353816-71acf1d0-7b57-4b33-94ab-4e57fbad1fa9.png)

_Main dashboard widget_ 

The `experimental` label appearss in small caps after `Ideas to write about based on unanswered searches`

![image](https://user-images.githubusercontent.com/73545194/131354073-e32496f1-3a6f-42d3-83c3-277e22844436.png)

- Confirmed that the styling and markup is consistent between all of the areas where the new component appears. - [Screenshot](https://user-images.githubusercontent.com/73545194/131354893-2f41e18d-854f-474e-9909-3fb1ea64abcd.png)
- Checked the new component on supported browsers and on smaller screen sizes.

**Comment by alex-moulin** [comment]
Experimental tag checked and approved from a UX pov.

@asvinb A quick note/question, I noticed that the background color is #F9F9F9 where on the designs it was #F1F3F4, was this changed for a reason as in most of the scenario, we can hardly see the background which makes the label look a bit floaty.

**Comment by aaemnnosttv** [comment]
@alex-moulin I don't think another color was picked intentionally but more that the closest color was chosen from our existing color palette. I agree that the contrast is a bit low though, although we don't have anything between that and `#e6e6e6` which is a bit too dark I think. I'm not opposed to adding another color but want to make sure that we do that with the whole design in mind. Right now we have some inconsistencies in some of the Figma files where things have more of a newer design and the plugin where things are largely unchanged from the initial release.

TL;DR I agree we should update it, but we might want to tweak the color based on the palette we're using now. I'll share all our colors with you separately to make sure that's the one we want 👍

**Comment by alex-moulin** [comment]
I see, my bad then, I picked a color from Google color palette as I couldn't find a specific palette for SK in Figma.
In that case, I think we could use $c-gallery: **#F0F0F0** to ensure we are not creating a new one and when we tackle the actual Design System we can review :)

**Comment by aaemnnosttv** [comment]
Sounds great, that looks good to me! Thanks @alex-moulin 👍

**Comment by samitron7** [comment]
Jumping in late but lgtm

**Comment by wpdarren** [comment]
### QA Update: ✅ 

- Can confirm that the background colour has been changed on all of the areas in the AC to #F0F0F0

![image](https://user-images.githubusercontent.com/73545194/131684609-4d76dafd-d51a-4990-8a56-4108a3a578cb.png)

**Comment by marrrmarrr** [comment]
When the screen size is slightly smaller, the "experimental" label is pushed to the line below and strangely aligned:
<img width="550" alt="Screen Shot 2021-09-09 at 16 40 32" src="https://user-images.githubusercontent.com/13304791/132708158-5f21d11b-dc14-4075-b2ae-a28203b768b8.png">

Is it possible to increase the space between the two lines and to align the label with the beginning of the text?

**Comment by felixarntz** [comment]
I've opened #4012 to address the above separately. Approval ✅
